romancal/multiband_catalog/multiband_catalog_step.py

```diff
@@ -183,12 +198,117 @@ def process(self, library):
         det_cat = det_catobj.catalog
         det_cat.meta["ee_fractions"] = {}
 
+        # Source Injection
```

Comment:
Can we move this to a new function, just to avoid process(...) getting so long?

Reply:
Refactored.